src/main/kotlin/me/partlysanestudios/partlysaneskies/data/cache/PetData.kt

```diff
@@ -0,0 +1,335 @@
+            return false
+        }
+        val upper = PartlySaneSkies.minecraft.currentScreen.getSeparateUpperLowerInventories()[0]?: return false
+        return upper.displayName.formattedText.removeColorCodes().contains("Pets")
+    }
+
+    // Using that list of pets, check to see if it's owned by a specific player
+    private fun getUsersPet(): Entity? {
+        val name = PartlySaneSkies.minecraft.thePlayer.getName()
+        val petEntities = getAllPets()
+        // If the pet says Ex: "[Lv100] *Su386*'s Black Cat" return that entity
+
+        for (entity in petEntities) {
+            if (entity.name.removeColorCodes().lowercase(Locale.getDefault()).contains(name.lowercase(Locale.getDefault()))) {
+                return entity
+            }
+        }
+        return null
+    }
+
+    // Gets all the pets current loaded by the game
+    private fun getAllPets(): List<Entity> {
+        val petEntities: MutableList<Entity> = ArrayList()
+        val armorStandEntities = getAllArmorStands()
+        val pattern = """§8\[§7Lv(\d+)§8] (§\w)([\w']+)\s*('s)? (\w+)""".toRegex()
+
+        // For every armor stand in the game, check if it's pet by looking for the level tag in front of the name.
+        // Ex: "*[Lv*100] Su386's Black Cat"
+        for (entity in armorStandEntities) {
+            if (pattern.find(entity.name) != null) {
+                petEntities.add(entity) // If so, add it to the list
+            }
+        }
+        return petEntities
+    }
+
+    // Gets all the armor stands currently loaded by the game
+    private fun getAllArmorStands(): List<Entity> {
+        val armorStandEntities: MutableList<Entity> = ArrayList()
+        val allEntities = getAllEntitiesInWorld()
+
+        // For every entity in the world, check if its instance of an armor stand
+        for (entity in allEntities) {
+            if (entity is EntityArmorStand) {
+                armorStandEntities.add(entity) // If so, add it to the list
+            }
+        }
+        return armorStandEntities
+    }
+
+    // Returns a list of all loaded entities in the world
+    private fun getAllEntitiesInWorld(): List<Entity> {
+        return PartlySaneSkies.minecraft.theWorld.getLoadedEntityList()
+    }
+
+    private fun String.getPetRarityFromColorCode(): Rarity {
+        return if (this == Rarity.COMMON.colorCode) {
+            Rarity.COMMON
+        } else if (this == Rarity.UNKNOWN.colorCode) {
+            Rarity.UNCOMMON
+        } else if (this == Rarity.RARE.colorCode) {
+            Rarity.RARE
+        } else if (this == Rarity.EPIC.colorCode) {
+            Rarity.EPIC
+        } else if (this == Rarity.LEGENDARY.colorCode){
+            Rarity.LEGENDARY
+        } else if (Rarity.MYTHIC.colorCode == this) {
+            Rarity.MYTHIC
+        } else {
+            Rarity.UNKNOWN
+        }
+    }
+
+    enum class Rarity(val order: Int, val colorCode: String): Comparable<Rarity> {
```

Comment:
This could be its own class, ULTIMATE_COSMETIC is also a rarity §4

Reply:
It's an enum for just pets, it should be called PetRarity but I thought with PetData.Rarity it would be fine

Reply:
We could do kotlin/.../data/Rarity.kt

Reply:
But it's not supposed to be for items and stuff like that, it's specifically for pet rarity

Reply:
I mean it could be a general rarity thing

Reply:
It could

Reply:
But do you wanna do that

Reply:
I'd much rather procrastinate it and do it three weeks later when we have to deal with it

Reply:
Most Su thing Ive ever heard

Reply:
If you want me to I can
I just don't want to

Reply:
I really want you to :(

Reply:
Urfghhhhh fine

Reply:
Done